core/controllers/acl_decorators.py

```diff
@@ -151,10 +151,18 @@ def test_does_classroom_exist(self, classroom_url_fragment, **kwargs):
             classroom_url_fragment)
 
         if not classroom:
-            _redirect_based_on_return_type(
-                self, '/learn/%s' % constants.DEFAULT_CLASSROOM_URL_FRAGMENT,
-                self.GET_HANDLER_ERROR_RETURN_TYPE)
-            return
+            # This decorator should only be used for JSON handlers, since all
+            # HTML page handlers are expected to be migrated to the Angular
+            # router and access validation for such pages should be done using
+            # the access validation handler endpoint.
+            if self.GET_HANDLER_ERROR_RETURN_TYPE == feconf.HANDLER_TYPE_JSON:
+                raise self.PageNotFoundException
+            else:
+                # As this decorator is not expected to be used with other
+                # handler types, raising an error here.
+                raise Exception(
+                    'does_classroom_exist decorator is expected to '
```

Comment:
Optional nit: "expected to be only used" --> "only expected to be used"

Reply:
changed.